### PR TITLE
Add App Map to App service/function view support

### PR DESCRIPTION
**Pull request by kryalama**
Add App Map to App service/function view support

### CODE REVIEW

**Comment by heyams** [comment]
do you need to send operating systems?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java [hunk 1]
```diff
@@ -52,6 +52,13 @@
 
   private static final String WEBSITE_HOME_STAMPNAME = "appSrv_wsStamp";
 
+  private static final String WEBSITE_OWNER_NAME = "appSrv_wsOwner";
+
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java:56]: does spotlessApply remove the empty line in between instance variables?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java [hunk 2]
```diff
@@ -98,6 +105,30 @@ public Boolean call() {
                 provider.addHeartBeatProperty(fieldName, websiteHomeStampName, true);
                 hasSetValues = true;
                 break;
+              case WEBSITE_OWNER_NAME:
+                String websiteOwnerName = getWebsiteOwnerName();
+                if (websiteOwnerName == null) {
+                  break;
+                }
+                provider.addHeartBeatProperty(fieldName, websiteOwnerName, true);
+                hasSetValues = true;
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java:114]:
```suggestion
                if (Strings.isNullOrEmpty(websiteOwnerName)) {
 provider.addHeartBeatProperty(fieldName, websiteOwnerName, true);
                hasSetValues = true;
                }
         
```

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java:114]: don't send when it's empty.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java [hunk 3]
```diff
@@ -98,6 +105,30 @@ public Boolean call() {
...
+                break;
+              case WEBSITE_RESOURCE_GROUP:
+                String websiteResourceGroup = getWebsiteResourceGroup();
+                if (websiteResourceGroup == null) {
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java:118]: same as above and please add empty check

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java [hunk 4]
```diff
@@ -98,6 +105,30 @@ public Boolean call() {
...
+                  break;
+                }
+                provider.addHeartBeatProperty(fieldName, websiteResourceGroup, true);
+                hasSetValues = true;
+                break;
+              case WEBSITE_SLOT_NAME:
+                String websiteSlotName = getWebsiteSlotName();
+                if (websiteSlotName == null) {
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java:126]: same here and please add empty check.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java (hunk 1 above)
Comment (kryalama) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/heartbeat/WebAppsHeartbeatProvider.java:56]: looks like it doesnot remove empty line.

**Review by heyams — APPROVED** [review]

**Pull request merged**